tools/src/block-gen/Options.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE ApplicativeDo   #-}
+{-# LANGUAGE QuasiQuotes     #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Command line options
+
+module Options
+       ( BlockGenOptions (..)
+       , getBlockGenOptions
+       ) where
+
+import           Universum
+
+import           Data.String.QQ               (s)
+import           Options.Applicative          (Parser, auto, execParser, footerDoc,
+                                               fullDesc, header, help, helper, info,
+                                               infoOption, long, metavar, option,
+                                               progDesc, strOption, switch, value)
+import           Text.PrettyPrint.ANSI.Leijen (Doc)
+
+data BlockGenOptions = BlockGenOptions
+    { bgoBlockN :: !Word32
+    -- ^ Number of blocks to generate.
+    , bgoNodesN :: !Word32
+    -- ^ Number of nodes.
+    , bgoPath   :: !FilePath
+    -- ^ Location of generated database.
+    , bgoAppend :: !Bool
+    -- ^ Whether to append to existing db.
+    , bgoSeed   :: !(Maybe Int)
+    }
+
+optionsParser :: Parser BlockGenOptions
+optionsParser = do
+    bgoBlockN <- option auto $
+        long    "blocks" <>
+        metavar "INT" <>
+        help "Length of blockchain."
+
+    bgoNodesN <- option auto $
+        long    "nodes" <>
+        metavar "INT" <>
+        help "Number of nodes."
+
+    bgoPath <- strOption $
+        long    "db-path" <>
+        metavar "FILEPATH" <>
+        value   "db-path" <>
```

Comment:
I think it's better to cool it `generated-db` or something like that by default. To make it clear that it was generated by the tool.